Enable to change clear cache option

paulosaure opened the pull request:
### New Pull Request Checklist

* ☑ I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* ☑ I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* ☑ I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found one : #940 (closed)

* ☑ I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* ☑ I have updated the documentation (if necessary)
* [ ] I have run the tests and they pass
* ☑ I have run the lint and it passes (`pod lib lint`)

This merge request fixes / reffers to the following issues: https://github.com/rs/SDWebImage/issues/2356

### Pull Request Description

According to https://github.com/rs/SDWebImage/issues/2356 issue, I would like to be able to change clear cache setting. 
I added the ability to clear cache by last accessed date not by creation date.

dreampiggy commented:
I approve this feature request ! Seems this is useful for people who need detail control for expiration filter.

There are some small little change to do before we merge this. Can you please check the review status again ? Thanks :)

If you have no time to do this, I can provide a help to create a new MR contain this feature as well...

paulosaure commented:
@dreampiggy 

Hello,
Thank you for your answer.

Enum `SDImageCacheExpireType` was not available in `SDImageCacheConfig.h` file, also I moved this enum to `SDImageCacheConfig.h` and I refactored the enum name to `SDImageCacheConfigExpireType`.

Is it good for you ?

dreampiggy commented:
@paulosaure Looks good. Another more change request, that `clearBy` property, seems not follow the  config type naming `SDImageCacheConfigExpireType`. Could we change that to `cacheExpireType`, which may be better ?

paulosaure commented:
Thanks for your feedback.

I fixed `cacheClearBy` property name and I easily understand your opinion. Have you any other suggestion  ?

dreampiggy commented:
@paulosaure Looks good now. I'll try to merge this to master after CI passed. :)

dreampiggy commented:
@paulosaure Merged. I add this to 4.4.2 milestone. We'll need maybe 1 weak to see if there are any other changes or bugfix for 4.4.2. And then we can release it.

paulosaure commented:
Awesome.
Thank you very much :D

paulosaure commented:
Hello @dreampiggy,

Do you know when will be the next release ?

dreampiggy commented:
@paulosaure Sorry here. Seems we find some bugs which I create a PR #2374 . But however, this need more time to test. 😢 

If you're in hurry. We can release a 4.4.2 soon, and fix that FLAnimatedImage issue a next patch version 4.4.3.

paulosaure commented:
It's better to minimize release number, let's see and wait.

Otherwise do you need help for your PR #2374 ?
